dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/options/src/com/ibm/ws/jaxrs/fat/options/OptionsResource.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jaxrs.fat.options;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+
+@Path("/test")
+public class OptionsResource {
+    @POST
+    public Response create() {
+        return Response.ok().build();
+    }
+
+    @GET
+    @Path("/{id}")
+    public Response get(@PathParam("id") String id) {
+        return Response.ok(id).build();
+    }
```

Comment:
This makes me wonder whether users would expect to see both `POST` and `GET` returned for the same OPTIONS request, since POST is available for "/test", but GET is only available for "/test/{id}".  

For example, if a user does "OPTIONS /test/123", if the available methods in the response header included POST, then that might make a user think they could send a request "POST /test/123", but in this case, that wouldn't work - but "GET /test/123" would.  Likewise, if they did "OPTIONS /test", if they get back POST and GET, they will only be able to do "POST /test" successfully.

The seems to imply that users should expect class- _and_ method-specific metadata from OPTIONS - it says (section 3.3.5):
```
On receipt of an OPTIONS request an implementation MUST either:
  1. Call a method annotated with a request method designator for OPTIONS or, if none present,
  2. GenerateanautomaticresponseusingthemetadataprovidedbytheJAX-RSannotationsonthematch-
ing class and its methods.
```

Reply:
Andy - I thought the same thing as I was coding it.  I will dig into making that fix.  Thanks